src/bottom-sheet.tsx

```diff
@@ -0,0 +1,748 @@
+                    dispatch('transitionEnd');
+                }, styles.transitionDuration);
+
+                return () => clearTimeout(tid);
+            }
+        }, [modalState]);
+
+        React.useEffect(() => {
+            if (modalState === 'closed') {
+                if (initRef.current) {
+                    onClose?.();
+                } else {
+                    dispatch('open');
+                }
+            } else {
+                initRef.current = true;
+            }
+        }, [modalState, onClose]);
+
+        const {onScroll, overlayStyle, ...dragableSheetProps} = useDraggableSheet({closeModal});
+
+        useSetModalStateEffect();
+        const bodyStyle = useLockBodyScroll();
+
+        if (modalState === 'closed') {
+            return null;
+        }
+
+        return (
+            <Portal>
+                <FocusTrap>
+                    {bodyStyle}
+                    {/* eslint-disable-next-line jsx-a11y/no-static-element-interactions */}
+                    <div
+                        className={classnames(styles.overlay, {
+                            [styles.closingOverlay]: modalState === 'closing',
+                        })}
+                        style={overlayStyle}
+                        onClick={closeModal}
+                    />
+                    <div
+                        className={classnames(styles.bottomSheetContainer, {
+                            [styles.closingSheet]: modalState === 'closing',
+                        })}
+                        onTransitionEnd={handleTransitionEnd}
+                        onAnimationEnd={handleTransitionEnd}
+                        {...dragableSheetProps}
+                        {...getPrefixedDataAttributes(dataAttributes, 'BottomSheet')}
+                        ref={ref}
+                    >
+                        <div className={styles.bottomSheet}>
+                            <div className={styles.bottomSheetContent}>
+                                <div className={styles.handleContainer}>
+                                    <div className={styles.handle} />
+                                </div>
+                                <section
+                                    role="dialog"
+                                    aria-modal="true"
+                                    aria-labelledby={modalTitleId}
+                                    onScroll={onScroll}
+                                    className={styles.children}
+                                >
+                                    {typeof children === 'function'
+                                        ? children({closeModal, modalTitleId})
+                                        : children}
+                                </section>
+                            </div>
+                            <div className={styles.modalCloseButton}>
+                                <IconButton size={32} onPress={closeModal} aria-label={texts.modalClose}>
```

Comment:
Is it possible to use the same hover effect as cards onClose icon in this element?
![image](https://github.com/Telefonica/mistica-web/assets/6722153/08028a16-0a34-4f9c-b8ea-35a689c3abd1)

Reply:
I can give it a try. But I'm afraid it's not easy to reutilize the same component, but I could copy/paste the styles

Reply:
maybe we could try to componentize IconButton with hover/press circle effect

Reply:
please include in the specs the size of the hover/active circle and the size of the x icon.
I have used 24px (default icon size) for the X, but in cards we have 32px for the circle and 20px for the X

Reply:
Updated to Icon: 20px / Container: 32px + color tokens for different states (both under a new comment in the component specs)

Reply:
done